src/main/java/org/apache/datasketches/kll/KllDoublesSketch.java

```diff
@@ -389,441 +271,81 @@ public double getQuantileLowerBound(final double fraction) {
-    final int adjPop = oddPop ? rawPop - 1 : rawPop;
-    final int halfAdjPop = adjPop / 2;
-
-    // level zero might not be sorted, so we must sort it if we wish to compact it
-    if (level == 0) {
-      Arrays.sort(items_, adjBeg, adjBeg + adjPop);
-    }
-    if (popAbove == 0) {
-      KllDoublesHelper.randomlyHalveUpDoubles(items_, adjBeg, adjPop, random);
-    } else {
-      KllDoublesHelper.randomlyHalveDownDoubles(items_, adjBeg, adjPop, random);
-      KllDoublesHelper.mergeSortedDoubleArrays(
-          items_, adjBeg, halfAdjPop,
-          items_, rawLim, popAbove,
-          items_, adjBeg + halfAdjPop);
-    }
-    levels_[level + 1] -= halfAdjPop;          // adjust boundaries of the level above
-    if (oddPop) {
-      levels_[level] = levels_[level + 1] - 1; // the current level now contains one item
-      items_[levels_[level]] = items_[rawBeg]; // namely this leftover guy
-    } else {
-      levels_[level] = levels_[level + 1];     // the current level is now empty
-    }
-
-    // verify that we freed up halfAdjPop array slots just below the current level
-    assert levels_[level] == rawBeg + halfAdjPop;
-
-    // finally, we need to shift up the data in the levels below
-    // so that the freed-up space can be used by level zero
-    if (level > 0) {
-      final int amount = rawBeg - levels_[0];
-      System.arraycopy(items_, levels_[0], items_, levels_[0] + halfAdjPop, amount);
-      for (int lvl = 0; lvl < level; lvl++) {
-        levels_[lvl] += halfAdjPop;
-      }
-    }
+  public void update(final double value) {  //possibly move proxy
+    updateDouble(value);
   }
 
-  private void addEmptyTopLevelToCompletelyFullSketch() {
-    final int curTotalCap = levels_[numLevels_];
-
-    // make sure that we are following a certain growth scheme
-    assert levels_[0] == 0; //definition of full
-    assert items_.length == curTotalCap;
-
-    // note that merging MIGHT over-grow levels_, in which case we might not have to grow it here
-    if (levels_.length < numLevels_ + 2) {
-      levels_ = KllHelper.growIntArray(levels_, numLevels_ + 2);
-    }
-
-    final int deltaCap = KllHelper.levelCapacity(k_, numLevels_ + 1, 0, m_);
-    final int newTotalCap = curTotalCap + deltaCap;
-
-    final double[] newBuf = new double[newTotalCap];
-
-    // copy (and shift) the current data into the new buffer
-    System.arraycopy(items_, levels_[0], newBuf, levels_[0] + deltaCap, curTotalCap);
-    items_ = newBuf;
-
-    // this loop includes the old "extra" index at the top
-    for (int i = 0; i <= numLevels_; i++) {
-      levels_[i] += deltaCap;
-    }
+  @Override //Used internally
+  double[] getDoubleItemsArray() { return doubleItems_; }
 
-    assert levels_[numLevels_] == newTotalCap;
+  @Override
+  double getDoubleItemsArrayAt(final int index) { return doubleItems_[index]; }
 
-    numLevels_++;
-    levels_[numLevels_] = newTotalCap; // initialize the new "extra" index at the top
-  }
+  @Override //Dummy
```

Comment:
what does this mean? it shouldn't be called? perhaps throw then?

Reply:
This is a limitation of the Java language, These studs are PkgPrivate and will never be called and only exist to satisfy the abstract interface specification at the root.  I didn't want to clutter the code with a whole bunch of throw exceptions.  Nonetheless, I will fix.

Done.